File: official-tutorial/src/main/java/com/bigdata/flink/proj/datset/Counter.java
package com.bigdata.flink.proj.datset;

import org.apache.hadoop.io.LongWritable;
import org.apache.hadoop.io.Text;
import org.apache.hadoop.mapred.JobConf;
import org.apache.hadoop.mapred.OutputCollector;
import org.apache.hadoop.mapred.Reducer;
import org.apache.hadoop.mapred.Reporter;

import java.io.IOException;
import java.util.Iterator;

/**
 * @author: huhao18@meituan.com
 * @date: 2021/5/15 4:08 下午
 * @desc:
 */
public class Counter implements Reducer<Text, LongWritable, Text, LongWritable> {
    @Override
    public void reduce(Text text, Iterator<LongWritable> iterator, OutputCollector<Text, LongWritable> outputCollector,
                       Reporter reporter) throws IOException {
        Long count = 0L;
        while (iterator.hasNext()) {
            count += iterator.next().get();
        }
        outputCollector.collect(text, new LongWritable(count));
    }

    @Override
    public void close() throws IOException {

    }

    @Override
    public void configure(JobConf jobConf) {

    }
}

File: official-tutorial/src/test/scala/com/bigdata/flink/proj/testing/taxi/scala/func/hadoop/Counter.scala
package com.bigdata.flink.proj.testing.taxi.scala.func.hadoop

/**
 * @author: huhao18@meituan.com
 * @date: 2021/5/15 4:14 下午 
 * @desc:
 *
 */

import org.apache.hadoop.io.{LongWritable, Text}
import org.apache.hadoop.mapred.{JobConf, OutputCollector, Reducer, Reporter}

import java.util
import scala.collection.convert.ImplicitConversions.`iterator asScala`

class Counter extends Reducer [Text, LongWritable, Text, LongWritable] {
  override def reduce(k2: Text, iterator: util.Iterator[LongWritable], outputCollector: OutputCollector[Text, LongWritable],
                      reporter: Reporter): Unit = {
    val value = iterator.map(_.get()).sum
    outputCollector.collect(k2,new LongWritable(value))
  }

  override def configure(jobConf: JobConf): Unit = {

  }

  override def close(): Unit = {

  }
}